index_create.go

```diff
@@ -0,0 +1,71 @@
+package opensearchtools
+
+import (
+	"io"
+	"time"
+)
+
+// CreateIndexRequest is a domain model union type for all the fields of CreateIndexRequests for all
+// supported OpenSearch versions.
+// Currently supported versions are:
+//   - OpenSearch 2
+//
+// An empty CreateIndexRequest will fail to execute. At least index is required to be added which matches to
```

Comment:
True, it will create an index with the default settings. For the empty request, probaly better to use the pointers for the optional fields to avoid that. I didn't use it before since it creates a lot of bloated code with many ifs.

Reply:
I don't think the ifs are that bloated, as they're valid checks. The other thing that's annoying with points on the primitive types is that consumers of the code have to declare a separate variable in order to use it.
```
duration := 1
req := CreateIndexRequest{Timeout: &duration}
```

Which adds, in my opinion, worse bloat to our consumers. I'd rather the library have a little bit of syntactical bloat than pass it on to consumers.